pyScienceMode2/sciencemode.py

```diff
@@ -118,66 +124,139 @@ def _get_last_ack(self, init: bool = False) -> bytes:
                 while not self.last_init_ack:
                     pass
                 last_ack = self.last_init_ack
+                self.info_received.append(last_ack)
                 self.last_init_ack = None
             else:
                 while not self.last_ack:
                     pass
                 last_ack = self.last_ack
+                self.info_received.append(last_ack)
                 self.last_ack = None
             return last_ack
         else:
             while 1:
                 packet = self._read_packet()
                 if packet and len(packet) != 0:
                     break
-            if self.show_log and packet[-1][6] in [t.value for t in self.Type]:
-                print(f"Ack received by rehastim: {self.Type(packet[-1][6]).name}")
-            return packet[-1]
+            if packet and not self.error_occured:
+                if self.show_log and packet[-1][6] in [t.value for t in self.Type]:
+                    print(f"Ack received by rehastim: {self.Type(packet[-1][6]).name}")
+                    self.info_received.append(packet[-1])
+                    self.return_list_ack()
+        if self.error_occured:
+            raise RuntimeError("Stimulation error")
+
+        return packet[-1]
+
+    def return_list_ack(self) -> list:
```

Comment:
Is it a private function ?
More explicit on the function name

Reply:
Yes, done.